core/src/test/scala/integration/kafka/api/SaslPlainPlaintextClientsWithInvalidCredentialsTest.scala

```diff
@@ -0,0 +1,122 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements. See the NOTICE
+  * file distributed with this work for additional information regarding copyright ownership. The ASF licenses this file
+  * to You under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the
+  * License. You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+  * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+  * specific language governing permissions and limitations under the License.
+  */
+package kafka.api
+
+import java.io.FileOutputStream
+import java.util.Locale
+import java.util.concurrent.{ExecutionException, Future}
+import scala.collection.JavaConverters.seqAsJavaListConverter
+
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerRecord, RecordMetadata}
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.errors.AuthenticationFailedException
+import org.apache.kafka.common.protocol.SecurityProtocol
+import org.junit.{After, Before, Ignore, Test}
+
+import kafka.admin.ConsumerGroupCommand.{ConsumerGroupCommandOptions, KafkaConsumerGroupService}
+import kafka.server.KafkaConfig
+import kafka.utils.{JaasTestUtils, TestUtils}
+
+class SaslPlainPlaintextClientsWithInvalidCredentialsTest extends BaseConsumerTest with SaslSetup {
```

Comment:
Would it be possible to avoid extending `BaseConsumerTest`? This would run all tests from `BaseConsumerTest`, which we probably don't want to do? If it is too big a change, then the current overriding of methods may be ok.